sw/host/opentitanlib/src/dif/lc_ctrl.rs

```diff
@@ -42,6 +44,35 @@ impl DifLcCtrlState {
         assert_eq!(value & 0b11111, value);
         (0..6).fold(0u32, |acc, _| (acc << 5) | value)
     }
+
+    pub fn parse_lc_state_str(lc_state_str: &str) -> Result<Self> {
```

Comment:
For your consideration:

1. implement `FromStr` here.
2. Enhance `with_unknown!` to implement `FromStr` for all `with_unknown!` enums.  For this case, however, the string form would end up being the UpperCamelCase form of the enumerator (ie: TestUnlocked0).  This may be at odd with what you're trying to convert from and I don't think rust macros can transform the stringified form from StudlyCaps to snake_case.
3. If (2), consider also having `with_unknown!` implement a `variants()` method which returns a list of all legal variants   as `&[&'static str]`.

Reply:
Yeah that was the problem I had, I originally implemented FromStr but then ran into the UpperCamelCase issue, because the loop in the build file generates snake_case.